src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -1037,6 +1037,14 @@ namespace winrt::TerminalApp::implementation
 
                 const auto progressValue = gsl::narrow<uint32_t>(state.Progress());
                 _tabStatus.ProgressValue(progressValue);
+
+                const auto color =
+                    (taskbarState == 2) ?
+                        winrt::Windows::UI::Colors::Red() :
+                        ((taskbarState == 4) ?
+                             winrt::Windows::UI::Colors::Orange() :
+                             winrt::Windows::UI::Colors::Green());

```

Comment:
I think I'd prefer a `switch` statement here for readability! 

Also for `tastbarState = 3` we should probably set it to the default color (it currently adopts the color based on the previous state, so it could be any of red/orange/green depending).